cbsearchx/search.go

```diff
@@ -586,6 +589,13 @@ func (h Search) DecodeCommonError(resp *http.Response) error {
 		err = ErrIndexExists
 	} else if strings.Contains(errText, "current index uuid") && strings.Contains(errText, "did not match input uuid") {
 		err = ErrIndexExists
+	} else if strings.Contains(errText, "unknown indextype") {
+		err = ErrUnknownIndexType
+	} else if strings.Contains(errText, "error obtaining vbucket count for bucket") ||
+		strings.Contains(errText, "requested resource not found") {
```

Comment:
I suspect this can be returned to us for other cases too?

Reply:
I'll add some more thorough tests but I don't think that it can. Only a missing source returns this error, since if we specify a Bucket/Scope name in the index definition that doesn't exist we get ErrIndexNotFound since the bucket and scope name are only used to construct the fully qualified index name.

Reply:
I can't find any other cases where this message is returned, the tests are in ING-677

Reply:
The error you get is: `search service not available`